### PR TITLE
Support remote file paths

### PR DESCRIPTION
**Pull request by colinking**
This allows you to create tasks without having a local task revision, f.e.:

```sh
airplane tasks deploy -f github.com/airplanedev/examples/node/hello-world-javascript/airplane.yml
```

### CODE REVIEW

**Review by joshma — APPROVED** [review]

File: pkg/taskdir/directory_github.go [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+	}
+
+	// TODO: consider using git 2.19's --filter option
+	// to select just the relevant subdirectory. However, this
+	// may not work with go-git.
+	//
+	// See: https://stackoverflow.com/questions/600079/how-do-i-clone-a-subdirectory-only-of-a-git-repository/52269934#52269934
+	r, err := git.PlainClone(tmpDir, false, &git.CloneOptions{
+		URL: fmt.Sprintf("https://github.com/%s/%s.git", fp.Org, fp.Repo),
+	})
```
[the remark this answers is not in the document]
Comment (colinking) [reply pkg/taskdir/directory_github.go:75]: That would require a GitHub API key vs. `git cloning` w/ standard git auth

File: pkg/taskdir/taskdef.go [hunk 2]
```diff
@@ -30,10 +34,42 @@ type Definition struct {
...
+	canPrompt := isatty.IsTerminal(os.Stdout.Fd())
+
+	if this.Slug == "" {
+		if !canPrompt {
+			return this, errors.New("Expected a slug")
+		}
+
+		if err := survey.AskOne(
+			&survey.Input{
+				Message: "Pick a slug to identify this task",
```
[the remark this answers is not in the document]
Comment (colinking) [reply pkg/taskdir/taskdef.go]: Went with a variant of that: `Pick a unique identifier (slug) for this task`

File: pkg/taskdir/directory_github.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (joshma) [reply pkg/taskdir/directory_github.go:75]: Ooh interesting. This assumes HTTP git auth is set up? Maybe we should support SSH and git URLs as well.

[the remark this answers is not in the document]
Comment (colinking) [reply pkg/taskdir/directory_github.go:75]: Think it's pretty standard to use https vs. ssh rewrites in your git config yeah?

[the remark this answers is not in the document]
Comment (joshma) [reply pkg/taskdir/directory_github.go:75]: Ah, maybe! I don't have that set up haha.

**Pull request merged**